panic when transpiling function pointer type definitions with no parameters; unknown node type: 'FunctionNoProtoType'

# Example C source:

## Working (no panic)

```c
typedef int (*foo)(void);
```

## Not working (panic)

Contents of `a.c`:
```c
typedef int (*foo)();
```

# Panic stack trace

Panic when running `c2go transpile a.c`

```
panic: unknown node type: 'FunctionNoProtoType 0x556e32bfde50 'int ()' cdecl'

goroutine 13 [running]:
github.com/elliotchance/c2go/ast.Parse(0xc00061076e, 0x31, 0x6afd19, 0x5)
```

# Additional information

Clang version:

```
u@x220 ~> clang -v
clang version 9.0.1 
Target: x86_64-pc-linux-gnu
```

`c2go` revision 4de3ec9.

--- a/ast/ast.go
+++ b/ast/ast.go
@@ -136,6 +136,8 @@ func Parse(fullline string) Node {
 		return parseFunctionDecl(line)
 	case "FullComment":
 		return parseFullComment(line)
+	case "FunctionNoProtoType":
+		return parseFunctionNoProtoType(line)
 	case "FunctionProtoType":
 		return parseFunctionProtoType(line)
 	case "ForStmt":

--- a/ast/function_no_proto_type.go
+++ b/ast/function_no_proto_type.go
@@ -1,0 +1,49 @@
+package ast
+
+// FunctionNoProtoType is a function type without parameters.
+//
+// Example:
+//    int (*)()
+type FunctionNoProtoType struct {
+	Addr        Address
+	Type        string
+	CallingConv string
+	ChildNodes  []Node
+}
+
+func parseFunctionNoProtoType(line string) *FunctionNoProtoType {
+	groups := groupsFromRegex(
+		"'(?P<type>.*?)' (?P<calling_conv>.*)",
+		line,
+	)
+
+	return &FunctionNoProtoType{
+		Addr:        ParseAddress(groups["address"]),
+		Type:        groups["type"],
+		CallingConv: groups["calling_conv"],
+		ChildNodes:  []Node{},
+	}
+}
+
+// AddChild adds a new child node. Child nodes can then be accessed with the
+// Children attribute.
+func (n *FunctionNoProtoType) AddChild(node Node) {
+	n.ChildNodes = append(n.ChildNodes, node)
+}
+
+// Address returns the numeric address of the node. See the documentation for
+// the Address type for more information.
+func (n *FunctionNoProtoType) Address() Address {
+	return n.Addr
+}
+
+// Children returns the child nodes. If this node does not have any children or
+// this node does not support children it will always return an empty slice.
+func (n *FunctionNoProtoType) Children() []Node {
+	return n.ChildNodes
+}
+
+// Position returns the position in the original source code.
+func (n *FunctionNoProtoType) Position() Position {
+	return Position{}
+}

--- a/ast/function_no_proto_type_test.go
+++ b/ast/function_no_proto_type_test.go
@@ -1,0 +1,18 @@
+package ast
+
+import (
+	"testing"
+)
+
+func TestFunctionNoProtoType(t *testing.T) {
+	nodes := map[string]Node{
+		`0x556e32bfde50 'int ()' cdecl`: &FunctionNoProtoType{
+			Addr:        0x556e32bfde50,
+			Type:        "int ()",
+			CallingConv: "cdecl",
+			ChildNodes:  []Node{},
+		},
+	}
+
+	runNodeTests(t, nodes)
+}

--- a/ast/position.go
+++ b/ast/position.go
@@ -409,9 +409,10 @@ func setPosition(node Node, position Position) {
 		n.Pos = position
 	case *TypedefType, *Typedef, *TranslationUnitDecl, *RecordType, *Record,
 		*QualType, *PointerType, *DecayedType, *ParenType,
-		*IncompleteArrayType, *FunctionProtoType, *EnumType, *Enum,
-		*ElaboratedType, *ConstantArrayType, *BuiltinType, *ArrayFiller,
-		*Field, *AttributedType:
+		*IncompleteArrayType, *FunctionNoProtoType, *FunctionProtoType,
+		*EnumType, *Enum, *ElaboratedType, *ConstantArrayType, *BuiltinType,
+		*ArrayFiller, *Field, *AttributedType:
+
 		// These do not have positions so they can be ignored.
 	default:
 		panic(fmt.Sprintf("unknown node type: %+#v", node))

